crates/mcp-server/src/router.rs

```diff
@@ -93,6 +95,15 @@ pub trait Router: Send + Sync + 'static {
         &self,
         uri: &str,
     ) -> Pin<Box<dyn Future<Output = Result<String, ResourceError>> + Send + 'static>>;
+    fn list_prompts(&self) -> Option<Vec<Prompt>> {
+        None
+    }
+    fn get_prompt(
+        &self,
+        _prompt_name: &str,
+    ) -> Option<Pin<Box<dyn Future<Output = Result<String, PromptError>> + Send + 'static>>> {
```

Comment:
nit: i'd remove the option here and leave these unimplemented for consistency? I suspect these are here so we don't break the existing servers, but i think we can go fix those up